### PR TITLE
Switch JS code to ES6

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Considering it's already quite big, I'll do the remaining files in another PR.

Part of #93058.

r? @notriddle

### CODE REVIEW

**Comment by Folyd** [comment]
Great. Is there something I can help switch JS code to ES6? 😸

**Comment by GuillaumeGomez** [comment]
It's already all done on my side. ^^'

**Comment by GuillaumeGomez** [comment]
@Folyd We will have some needs after this first wave. Let's talk about it so you know what remains to be done after. :)

**Comment by Folyd** [comment]
> @Folyd We will have some needs after this first wave. Let's talk about it so you know what remains to be done after. :)

Cool. I'm just an `@` away. 👏

File: src/librustdoc/html/static/js/settings.js [hunk 1]
```diff
@@ -1,4 +1,7 @@
 // Local js definitions:
+/* eslint-env es6 */
+/* eslint no-var: "error" */
+/* eslint prefer-const: "error" */
 /* global getSettingValue, getVirtualKey, onEachLazy, updateLocalStorage, updateSystemTheme */
 /* global addClass, removeClass */
```
Comment (bjorn3) [inline src/librustdoc/html/static/js/settings.js:7]: Should this and the other javascript files use `"use strict";`?

Comment (GuillaumeGomez) [reply src/librustdoc/html/static/js/settings.js:7]: It's mentioned in the issue already. I'll let this part of other contributors alongside adding more eslint checks.

**Pull request merged**